src/bin/solang.rs

```diff
@@ -293,6 +337,18 @@ fn main() {
         }
         Some(("compile", matches)) => compile(matches),
         Some(("doc", matches)) => doc(matches),
+        Some(("metadata", matches)) => {
+            let target = target_arg(matches);
+
+            // For Ethereum, we can generate an interface from an abi file
+            // For Substrate, we can generate an interface from a contract metadata file
+            if target == Target::Solana {
+                metadata::idl(matches)
+            } else {
+                eprintln!("error: metadata for target {} not supported", target);
+                exit(2);
```

Comment:
Shouldn't this be `exit(1)`?

Reply:
`exit(2)` means bad usage (i.e. bad command line) by convention.